BUG: build panic: runtime error: slice bounds out of range [:19] with length 0

### Sealos Version

v4.1.3

### How to reproduce the bug?

github action build cloud image error , i meet once at local.
https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402
```
Run .github/scripts/apps.sh
2022-11-07T09:36:24 info lookup in path charts
2022-11-07T09:36:24 info sub chart is pxc-db
2022-11-07T09:36:25 info sub chart is pxc-operator
2022-11-07T09:36:25 warn if you access private registry,you must be 'sealos login' or 'buildah login'
2022-11-07T09:36:25 info pull images [docker.io/library/percona:8.0 docker.io/percona/percona-xtradb-cluster-operator:1.11.0 docker.io/percona/percona-xtradb-cluster-operator:1.11.0-haproxy docker.io/percona/percona-xtradb-cluster-operator:1.11.0-logcollector docker.io/percona/percona-xtradb-cluster-operator:1.11.0-pxc8.0-backup docker.io/percona/percona-xtradb-cluster:8.0.27-18.1 docker.io/percona/pmm-client:2.28.0 percona/percona-xtradb-cluster-operator:1.11.0 percona/percona-xtradb-cluster-operator:1.11.0-haproxy percona/percona-xtradb-cluster-operator:1.11.0-logcollector percona/percona-xtradb-cluster:8.0.27-18.1] for platform is linux/amd64
Pulling image: docker.io/library/percona:8.0
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0-haproxy
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0-logcollector
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0-pxc8.0-backup
Pulling image: docker.io/percona/percona-xtradb-cluster:8.0.27-18.1
Pulling image: docker.io/percona/pmm-client:2.28.0
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0-haproxy
Pulling image: docker.io/percona/percona-xtradb-cluster-operator:1.11.0-logcollector
Pulling image: docker.io/percona/percona-xtradb-cluster:8.0.27-18.1
e4e8bbae6e34: Download complete
62fa042e4cf7: Download complete
deda48287c30: Download complete
96dee67348b3: Download complete
0344366a246a: Download complete
444[12](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:13)781f773: Download complete
0344366a246a: already exists
3f0f2255cb70: Download complete
5665c67922cd: Download complete
9b3d96517551: Download complete
0344366a246a: already exists
4752687a61a9: Download complete
5665c67922cd: already exists
4752687a61a9: already exists
0344366a246a: already exists
4752687a61a9: Download complete
4752687a61a9: Download complete
4752687a61a9: already exists
0344366a246a: already exists
19bee1074fa5: Download complete
691fed3dc6b3: Download complete
fd57752454d6: Download complete
3bc51a6b094c: Download complete
b5984fcc1917: Download complete
fa727e[15](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:16)1cc1: Download complete
fd57752454d6: Download complete
panic: runtime error: slice bounds out of range [:19] with length 0

goroutine 310 [running]:
github.com/labring/sealos/pkg/registry.(*DefaultImageSaver).saveBlobs.func2()
	github.com/labring/sealos/pkg/registry/save.go:331 +0xb54
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.0.0-202206011502[17](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:18)-0de741cfad7f/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.0.0-[20](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:21)220601150[21](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:22)7-0de7[41](https://github.com/labring/cluster-image/actions/runs/3409154505/jobs/5670576402#step:4:42)cfad7f/errgroup/errgroup.go:72 +0xa5
Error: Process completed with exit code 2.
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Operating environment

```markdown
- Sealos version:
- Docker version:
- Kubernetes version:
- Operating system:
- Runtime environment:
- Cluster size:
- Additional information:
```


### Additional information

_No response_

--- a/pkg/registry/save.go
+++ b/pkg/registry/save.go
@@ -326,6 +326,9 @@ func (is *DefaultImage) saveBlobs(imageDigests []digest.Digest, repo distributio
 	ls, _ := blobStore.(localStore).Local(is.ctx)
 	for _, blob := range blobLists {
 		tmpBlob := blob
+		if len(blob) == 0 {
+			continue
+		}
 		numCh <- struct{}{}
 		eg.Go(func() error {
 			defer func() {

